### PR TITLE
`ip_range` aggregation should accept null bounds.

### LINKED ISSUE #21006: IP range agg from/to params no longer accept null
**Elasticsearch version**: 6.0, 5.0

**Plugins installed**: none

**JVM version**: java version "1.8.0_60"

**OS version**: OSX El Capitan

**Description of the problem including expected versus actual behavior**:

An IP range agg with `from: null` or `to: null` will respond with the following error: 

```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_state_exception",
                "reason": "Can't get text on a VALUE_NULL at 1:261"
            }
        ],
        "type": "illegal_state_exception",
        "reason": "Can't get text on a VALUE_NULL at 1:261"
    },
    "status": 500
}
```

This is most likely the intended behavior, but 2.x was more lenient and allowed null and there's no mention of this in the breaking change docs as far as I can see. So I'm guessing the correct solution is simply to document this change.

### PR DESCRIPTION
**Pull request by jpountz**
Closes #21006

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/40_range.yaml [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+  - do:
+      search:
+        body: { "size" : 0, "aggs" : { "ip_range" : { "ip_range" : { "field" : "ip", "ranges": [ { "to": "192.168.0.0" }, { "from": "192.168.0.0", "to": "192.169.0.0" }, { "from": "192.169.0.0" } ] } } } }
+
+  - match: { hits.total: 3 }
+
+  - length: { aggregations.ip_range.buckets: 3 }
+
+#  - match: { aggregations.ip_range.buckets.0.key: "*-192.168.0.0" }
```
Comment (cbuescher) [inline rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/40_range.yaml]: Are these assertions that are commented out a reminder that this _should_ work at some point? If so, is there an issue for this that could be linked to in a comment? That way someone doesn't have to wonder if this is a left-over or a TODO.

**Review by cbuescher — APPROVED** [review]
LGTM but left one question regarding some commented lines in the tests

File: rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/40_range.yaml (hunk 1 above)
Comment (cbuescher) [inline rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/40_range.yaml]: I saw the issue now (#21045), I'd just put a comment in the test so this is clear immediately

**Comment by jpountz** [comment]
@cbuescher I pushed a commit with a comment.

**Pull request merged**